src/Features/Core/Portable/UnusedReferences/UnusedReferencesRemover.cs

```diff
@@ -274,11 +309,22 @@ internal static ImmutableArray<string> GetAllCompilationAssemblies(ReferenceInfo
                     continue;
                 }
 
-                await referenceCleanupService.TryUpdateReferenceAsync(
-                    projectFilePath,
-                    referenceUpdate,
-                    cancellationToken).ConfigureAwait(true);
+                try
+                {
+                    var success = await referenceCleanupService.TryUpdateReferenceAsync(
```

Comment:
this feels odd.  the moment we fail, we switch over to returning 'false'.  but we keep on going in this loop.  why not just return false on the first failure?  (this needs doc'ing)/

Reply:
reverting this bit and moving to a separate pr.